presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -221,11 +225,17 @@ public AbstractOrcRecordReader(
                     stripes.add(stripe);
                     stripeFilePositions.add(fileRowCount);
                     totalRowCount += stripe.getNumberOfRows();
+                    if (!startRowPosition.isPresent()) {
```

Comment:
Not sure what exactly you are doing here. Are you trying to calculate which row at the file level is the starting row?

I suggest you to add test cases  for reading all stripes in the file, and a test case for reading some stripes that are not starting from the first stripe.